src/request/components/request-detail-panel-logging.tsx

```diff
@@ -2,10 +2,71 @@
 
 import { FontAwesomeIcon } from '../../shell/components/FontAwesomeIcon';
 import { ILogMessage } from '../messages/ILogMessage';
-import { ILoggingComponentModel, ILoggingLevelModel } from '../component-models/ILoggingComponentModel';
+import { ILoggingComponentModel, ILoggingLevelModel, ILogMessageModel } from '../component-models/ILoggingComponentModel';
 
 import _ = require('lodash');
 import React = require('react');
+import Highlight = require('react-highlight');
+
+interface ILogMessageProps {
+    message: ILogMessageModel;
+}
+
+interface ILogMessageState {
+    isExpanded: boolean;
+}
+
+class LogMessage extends React.Component<ILogMessageProps, ILogMessageState> {
+    constructor(props?) {
+        super(props);
+
+        this.state = {
+            isExpanded: false
+        };
+    }
+
+    public render() {
+        return (
+            <div className='tab-logs-table-message'>
+                <div className={this.getIconClass()} onClick={e => this.onToggleExpansion()}><FontAwesomeIcon path={this.getIconPath()} /></div>
+                {
+                    (this.state.isExpanded && this.props.message.isObject)
+                        ? <div className='tab-logs-table-message-object'><Highlight language='javascript'>{this.props.message.message}</Highlight></div>
+                        : <div className={this.getMessageClass()}>{this.props.message.spans.map(span => <span className={span.wasReplaced ? 'tab-logs-table-message-replaced-region' : ''}>{span.text}</span>)}</div>
+                }
+            </div>);
+    }
+    
+    private onToggleExpansion(): void {
+        this.setState({
+            isExpanded: !this.state.isExpanded
+        });
+    }
+
+    private getIconClass() {
+        if (this.state.isExpanded) {
+            return 'tab-logs-table-message-icon-expandable';
+        }
+        else if (this.props.message.isObject) {
+            return 'tab-logs-table-message-icon-expandable';
+        }
+        else {
+            return 'tab-logs-table-message-icon';
+        }
+    }
+
+    private getMessageClass() {
+        return this.state.isExpanded
+            ? 'tab-logs-table-message-multiline'
+            : '';
+    }
```

Comment:
>is used to style the plain-text element in both collapsed

Seems a little weird then in terms of logic. They way I read this, is that if the expanded, then apply the multiline class.